### PR TITLE
feat: transform GitHub alert blocks into Docusaurus admonitions

### LINKED ISSUE #570: Transform GitHub alert blocks into Docusaurus admonitions
GitHub released a new "Alerts" syntax in GitHub-flavoured Markdown (GFM) for highlighting blocks in late 2022:

> [!NOTE]
> This is an important note!

https://github.com/orgs/community/discussions/16925

These are a non-standard Markdown extension, and not natively supported by Docusaurus (ref https://github.com/facebook/docusaurus/issues/7471). However, Docusaurus has its own [admonition syntax](https://docusaurus.io/docs/markdown-features/admonitions) that serves approximately the same purpose:

```
:::note

This is an important note!

:::
```

One of our goals with Electron's documentation is to make the code readable in both GitHub.com and Docusaurus contexts. Ideally, we'd be able to transform the GitHub Alert syntax into Docusaurus admonitions to improve this interop.

There seems to be an open-source Remark plugin that we could try out to this end: https://github.com/incentro-dc/remark-github-admonitions-to-directives

### PR DESCRIPTION
**Pull request by AmrMustafa282**
# fixed #570 
- added Remark Github Alerts to Directives plugin to docusaurus.config.ts
- configure the plugin in docs and blog for future need
## preview after updates
Screenshot 2024-10-12 at 3 05 23 AM

### CODE REVIEW

**Review by AmrMustafa282 — COMMENTED** [review]
initial review

File: package.json [hunk 1]
```diff
@@ -45,6 +45,8 @@
     "react-dom": "^18.3.1",
     "react-markdown": "^9.0.1",
     "react-useportal": "^1.0.14",
+    "remark-directive": "^3.0.0",
```
Comment (erickzhao) [inline package.json:48]: question: why do we need to install `remark-directive` separately here?

Comment (erickzhao) [reply package.json]: [165b808](https://github.com/electron/website/pull/655/commits/165b808a25752b20b80a4e6fa103e4e2321b7688)

**Review by erickzhao — APPROVED** [review]
tweaked a few things and re-landed lockfile changes according to our policies

**Review by VerteDinde — APPROVED** [review]
LGTM (Erick changed the yarn.lock, so we're good there 😄 )

**Pull request merged**